### PR TITLE
machine/samd21: switch UART to use new pin configuration

### PR DESCRIPTION
**Pull request by aykevl**
This allows all possible UART pin configurations to be used and avoids some tricky configuration.

Will rebase on top of dev once #597 lands.

### CODE REVIEW

File: src/machine/board_circuitplay_express.go [hunk 1]
```diff
@@ -70,10 +70,10 @@ const (
 
 // UART1 on the Circuit Playground Express.
 var (
-	UART1 = UART{Bus: sam.SERCOM1_USART,
+	UART1 = UART{
 		Buffer: NewRingBuffer(),
-		Mode:   PinSERCOM,
-		IRQVal: sam.IRQ_SERCOM1,
+		Bus:    sam.SERCOM4_USART,
+		SERCOM: 4,
```
Comment (aykevl) [inline src/machine/board_circuitplay_express.go]: The pin configuration here (`PB08` / `PB09`) was never supported and wouldn't work on SERCOM1 anyway. Switched to SERCOM4 keeping the pin numbers, following the common pinout of the Circuit Playground Express.

File: src/machine/board_trinket.go [hunk 2]
```diff
@@ -44,10 +44,10 @@ const (
 
 // UART1 on the Trinket M0.
 var (
-	UART1 = UART{Bus: sam.SERCOM1_USART,
+	UART1 = UART{
 		Buffer: NewRingBuffer(),
-		Mode:   PinSERCOM,
-		IRQVal: sam.IRQ_SERCOM1,
+		Bus:    sam.SERCOM0_USART,
+		SERCOM: 0,
```
Comment (aykevl) [inline src/machine/board_trinket.go]: It appears that using SERCOM1 was a bug: the only SERCOM that is supported on the TX/RX pins is SERCOM0. Previous UART support wouldn't have worked on the Trinket M0 TX/RX pins anyway (`PA06`/`PA07`). This PR fixes that, and should thereby add support for UART on the Trinket M0.

**Comment by aykevl** [comment]
Confirmed that the espconsole example still works with this PR. I haven't done any other testing other than cross-checking all configuration as in the self-review above.

**Comment by aykevl** [comment]
Rebased.

**Comment by deadprogram** [comment]
I also did some similar testing with this and worked as expected. Now merging, thank you @aykevl

**Pull request merged**